pkg/api/write.go

```diff
@@ -151,3 +192,65 @@ func buildWriteError(w http.ResponseWriter, err string) {
 	log.Error("msg", "Write header validation error", "err", err)
 	http.Error(w, err, http.StatusBadRequest)
 }
+
+type importPayload struct {
+	Labels  map[string]string `json:"labels"`
+	Samples []importSample    `json:"samples"`
+}
+
+type importSample struct {
+	Timestamp int64
+	Value     float64
+}
+
+func (i *importSample) UnmarshalJSON(data []byte) error {
+	var (
+		v  []interface{}
+		ok bool
+	)
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	if len(v) != 2 {
+		return fmt.Errorf("invalid sample, should contain timestamp and value: %v", v)
+	}
+
+	floatTimestamp, ok := v[0].(float64)
+	if !ok {
+		return fmt.Errorf("cannot convert timestamp to int64 type for sample: %s", data)
+	}
+	i.Timestamp = int64(floatTimestamp)
```

Comment:
can we make sure the floatTimestamp has no decimal part. We don't want to silently drop precision here as that implies the input is wrong